profiler/profiler.py

```diff
@@ -3,129 +3,127 @@
 import os
 import re
 import subprocess
-from datetime import date, datetime
+import sys
+from datetime import datetime
 from subprocess import PIPE
-from typing import Dict, Optional
+from sys import stderr
+from typing import Any, Dict, Optional
 
 import somacore
 
 import tiledbsoma
-from data import FileBasedProfileDB, ProfileData, ProfileDB
 
-TILEDB_STATS_FILE_PATH = "./tiledb_stats.txt"
-
-
-def read_context() -> Dict[str, str]:
-    """Run the context generator process and collects the JSON printed output"""
-    context_process = subprocess.Popen(
-        ["python", "context_generator.py"], stdout=PIPE, stderr=PIPE
-    )
-    stdout, stderr = context_process.communicate()
-    stats = json.loads(stdout.decode("utf-8"))
-    return stats
+from .context_generator import host_context
+from .data import FileBasedProfileDB, ProfileData, ProfileDB
+
+GNU_TIME_FORMAT = (
```

Comment:
Why do we need both `GNU_TIME_FORMAT` and `GNU_TIME_FORMAT`?

Reply:
Assuming you mean why do we need both `GNU_TIME_FORMAT` and `GNU_TIME_OUTPUT_REGEXP`. The former is for parsing the output, and the latter is for defining the output. We needed a custom output format to get elapsed (wall clock) time stat as a seconds value (not HH:MM:SS), which simplified things. The elapsed time format differs if the value exceeds an hour, which I found out the hard way. :) So I chose to just output as seconds for simplicity.

Now that we can control the output format, I could have moved to a simpler-to-parse format, but stuck with nearly the same format for now.